src/uarch-step.cpp

```diff
@@ -1089,34 +1091,34 @@ static inline void executeInsn(UarchState &a, uint32 insn, uint64 pc) {
 }
 
 template <typename UarchState>
-uarch_step_status uarch_step(UarchState &a) {
+UArchStepStatus uarch_step(UarchState &a) {
     // This must be the first read in order to match the first log access in machine::verify_state_transition
     uint64 cycle = readCycle(a);
     // do not advance if cycle will overflow
     if (cycle == UINT64_MAX) {
-        return uarch_step_status::cycle_overflow;
+        return UArchStepStatus::CycleOverflow;
     }
     // do not advance if machine is halted
     if (readHaltFlag(a)) {
-        return uarch_step_status::uarch_halted;
+        return UArchStepStatus::UArchHalted;
     }
     // execute next instruction
     auto pc = readPc(a);
     auto insn = readUint32(a, pc);
```

Comment:
`auto` still being used in the above lines.

Reply:
Good catch! @mpernambuco you probably missed the second patch I sent you, could you add it back?